scania-digital-design-system/sdds
# AB#252 Improvement/separate assets

**Describe pull-request**  
- Separate icons into another package
- Use CDN for image and fonts (Remove base64 solution)
- Use dynamic import for adding the icons via sdds-icons
- Icons now available as webfont (can be used from local package or from CDN)
- Enable source-mapping in theme-light only on dev environment (on npm start). BUt disable it on build (on npm run build).

**Solving issue**  
_Add which issue this pull-request solves by adding # plus the number of the issue_
Fixes: #129  [AB#252](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/252)

**How to test**  

1. Check if icons package work
  - Go to sdds/theme/core/icons
  - Run `npm run build`
  - Build @scania/components and @scania/theme-light as well.
  - Test solution in HTML demo (I made a page called icons.html, linking to all 3 packages - icons, theme, component), this page shows the webfont
  - Check in storybook that all components are okay (including header, footer, navigation)
  - You can compare to the latest build of storybook in netlify https://deploy-preview-139--sdds-storybook.netlify.app/?path=/story/patterns-footer--with-social-links
  - Test solution in angular (use `npm link` in sdds-website, linking to all 3 packages) - Use this branch https://github.com/scania-digital-design-system/sdds-website/tree/improvement/theme_light_optimized and check in Foundation > Pictograms page

2. Check if theme light are not having issue with new solution (Pay attention to logotypes, symbol, and fonts)


**Additional context**  

- **Need to update CDN link when the URL is ready (waiting for DNS registration)** DONE ✔️ 
- Will need to update c-icon to sdds-icon in site, branches created, but PR can be made when package is released
https://github.com/scania-digital-design-system/sdds-website/tree/improvement/theme_light_optimized

- Documentation has been added in the CMS
- c-cookie, c-footer, and c-modal has dependency to c-icon, it is replaced with sdds-icon now, but application need to import icons package to make it work. (Should we care about old c- components and add the svg code instead?)
